### ISSUE TITLE
catkin build argument parsing error on leading "l" 

### ISSUE THREAD

**Issue by astumpf** [catkin/catkin_tools#516]
[//]: # (Thank you for reporting an issue for catkin_tools!)

[//]: # (Please fill out the details below so that your issue can be resolved quickly!)
[//]: # (If you want to suggest a feature or start a discussion, feel free to delete them.)

### System Info

* Operating System: Linux ThinkPad-T440p 4.13.0-41-generic #46~16.04.1-Ubuntu SMP Thu May 3 10:06:43 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
* Python Version: Python 2.7.12
* Version of catkin_tools: catkin_tools 0.4.4 (C) 2014-2018 Open Source Robotics Foundation
catkin_tools is released under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
---
Using Python 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
* ROS Distro: Kinetic Kame

### Build / Run Issue

[//]: # (If you are migrating from catkin_make, please follow the migration guide:)
[//]: # (http://catkin-tools.readthedocs.org/en/latest/migration.html)

[//]: # (Please also check for solved issues here:)
[//]: # (http://catkin-tools.readthedocs.org/en/latest/troubleshooting.html)

[//]: # (And check for open issues here:)
[//]: # (https://github.com/catkin/catkin_tools/issues?q=is%3Aopen+is%3Aissue+label%3Abug)

[//]: # (If you have an issue building and running code, please verify the following:)

* [ ] Works with `catkin_make`
* [ ] Works with `catkin_make_isolated --merge`
* [ ] Works with `catkin build`
* [ ] Works with `catkin build -p1`

### Expected Behavior

[//]: # (Please describe what you expected or think should happen.)
catkin build should complete with a regular build process.

### Actual Behavior

[//]: # (Please describe what actually happens.)

[//]: # (Please put long CLI output into Gists: http://gist.github.com)
[//]: # (Use `VERBOSE=1 catkin build -p1 -v` for verbose output)
[//]: # (Please post `logs` directory if available.)
Argument parsing went wrong. Instead of taking the full name package name it parses a preceding "l" directly followed by a number as an argument leading to a wrong package name. 

### Steps to Reproduce the Issue

[//]: # (Please give a series of command-line commands which cause the problem.)
[//]: # (Include a checkout URI for the workspace so that we can reproduce the issue.)
[//]: # (If we can't reproduce it on our own machines, we can't fix it!)

Just try to run catkin build on a specific package name pattern with a preceding "l" and number, e.g. "l1". The package must actually not exist to reproduce the error.
```bash
catkin build l1_test
```
results in 
```bash
[...]
Additional Make Args:        l1
[...]
----------------------------------------------------
[build] Found '157' packages in 0.0 seconds.                                                                                                                                                               
[build] Given package '_test' is not in the workspace
```

**Comment by jbohren-hbr** [comment]
Problem is this regex: https://github.com/catkin/catkin_tools/blob/c60687f6c06a39f848c76fae3aefec6af50a696a/catkin_tools/argument_parsing.py#L229-L231

**Comment by jbohren-hbr** [comment]
It's capturing `-lX` as a load average limit of `X`

**Comment by jbohren-hbr** [comment]
@astumpf would you be willing to contribute a patch to fix this?

**Comment by timonegk** [comment]
Fixed in #573.
